simde/x86/sse2.h

```diff
@@ -6590,7 +6590,7 @@ simde_mm_slli_epi16 (simde__m128i a, const int imm8)
   #define simde_mm_slli_epi16(a, imm8) \
     ((imm8 & ~15) ? simde_mm_setzero_si128() : simde__m128i_from_altivec_i16(vec_sl(simde__m128i_to_altivec_i16(a), vec_splat_u16(HEDLEY_STATIC_CAST(unsigned short, imm8)))))
 #elif defined(SIMDE_LOONGARCH_LSX_NATIVE)
-  #define simde_mm_slli_epi16(a, imm8) ((imm8 & ~15) ? simde_mm_setzero_si128() : simde__m128i_from_lsx_i64(__lsx_vslli_h(simde__m128i_to_private(a).lsx_i64, ((imm8) & 15))))
+  #define simde_mm_slli_epi16(a, imm8) ((imm8 & ~15) ? simde_mm_setzero_si128() : simde__m128i_from_lsx_i64(__lsx_vsll_h(simde__m128i_to_private(a).lsx_i64, __lsx_vreplgr2vr_h((imm8) & 15))))
```

Comment:
Why are you changing this? For `simde_mm_slli_epi16` we require `imm8` to be a constant value.

See https://github.com/simd-everywhere/simde/pull/1263/commits/1715c161b794c5e4aef960c94647fa4f4ac5ab3d#diff-d5773576112e65731ce8f01f5beee4f665e6fcffd1017ee76c583fa0988a40b4R6554-R6555

Same for `simde_mm_slli_epi32`, `simde_mm_slli_epi64`, `simde_mm_srli_epi16`, `simde_mm_srli_epi32` and `simde_mm_srli_epi64`

Reply:
Wacky. Sure, it doesn't bother me either way

Reply:
```
#include <stdio.h>
#include "/data/home/Jin/data/community/simde-loong64/simde/x86/sse2.h"

int main()
{
        simde__m128i vec0;
        const int imm8 = 1;     //emulate use-cases that end-users pass variable for shift count.
        simde_mm_slli_epi16(vec0, imm8);
	return 0;
}

```
I built above demo on real loongarch hardware with simde git master, and the compiler output errors:
![image](https://github.com/user-attachments/assets/ccbb2187-8387-4601-bcbe-d45a601c6320)
Same for simde_mm_slli_epi32, simde_mm_slli_epi64, simde_mm_srli_epi16, simde_mm_srli_epi32 and simde_mm_srli_epi64

Reply:
```
#include <stdio.h>
#include "/data/home/Jin/data/community/simde-loong64/simde/x86/sse2.h"

int main()
{
        simde__m128i vec0;
        //const int imm8 = 1;     //emulate use-cases that end-users pass variable for shift count.
        simde_mm_slli_epi16(vec0, 1);
	return 0;
}
```
If user pass immediate values for shift count just like above demo,  it's ok for loongarch compiler.

Reply:
So, at present, we need to load imm8 into vectors and then use sll version instruction instead. If in the feature, the loongarch compiler can accept a variable for shift count for slli instruction then we can solve this dilemma.